src/analog.cpp

```diff
@@ -0,0 +1,31 @@
+#include "analog.h"
+
+// Ticker to poll input readings at a fixed rate
+STM32TimerInterrupt dataUpdater(TIM2);
+
+void initAnalog() {
+    initADC(ADC1);
+
+    // start the Ticker to periodically read IO
+    if(dataUpdater.attachInterruptInterval(IO_UPDATE_PERIOD, readAnalog)) {
+        printf("starting readAnalog timer\n");
+    } else {
+        printf("problem starting readAnalog timer\n");
+    }
+}
+
+void readAnalog() {
+    lv_12V_telem = readADC(ADC_CHANNEL_6) * 3.3 *  35.1/5.1; // PA_1
+    lv_5V_telem = readADC(ADC_CHANNEL_12) * 3.3 * 15.1/5.1; // PA_7
+    lv_5V_current = readADC(ADC_CHANNEL_15); // PB_0
```

Comment:
This reads the output voltage from the INA, but we need to convert it to a current value. You will probably have to look at the schematics and [INA datasheet](https://www.ti.com/lit/ds/symlink/ina180-q1.pdf?HQS=dis-dk-null-digikeymode-dsf-pf-null-wwe&ts=1738108718939) to figure out what the exact conversion is. John Choi from the LV team might have more insights on how this works too.